dotnet/src/Functions/Functions.Markdown/Extensions/KernelFunctionsMarkdownExtensions.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.IO;
+using Microsoft.Extensions.Logging;
+using Microsoft.SemanticKernel.Functions.Markdown.Functions;
+using Microsoft.SemanticKernel.Models;
+using Microsoft.SemanticKernel.TemplateEngine;
+
+namespace Microsoft.SemanticKernel.Functions.Markdown.Extensions;
+
+/// <summary>
+/// Class for extensions methods to define functions using prompt markdown format.
+/// </summary>
+public static class KernelFunctionsMarkdownExtensions
+{
+    /// <summary>
+    /// Creates an <see cref="ISKFunction"/> instance for a semantic function using the specified markdown text.
+    /// </summary>
+    /// <param name="kernel">Kernel instance</param>
+    /// <param name="resourceName">Resource containing the YAML representation of the <see cref="PromptModel"/> to use to create the semantic function</param>
+    /// <param name="functionName">The function name</param>
+    /// <param name="pluginName">The optional name of the plug-in associated with this method.</param>
+    /// <param name="promptTemplateFactory">>Prompt template factory.</param>
+    /// <returns>The created <see cref="ISKFunction"/>.</returns>
+    public static ISKFunction FromMarkdownResource(
+        this IKernel kernel,
+        string resourceName,
+        string? functionName = null,
+        string? pluginName = null,
+        IPromptTemplateFactory? promptTemplateFactory = null)
+    {
+        functionName ??= Path.GetFileNameWithoutExtension(resourceName);
+        return KernelFunctionMarkdown.FromPromptMarkdownResource(resourceName, functionName, pluginName, promptTemplateFactory, kernel.LoggerFactory);
+    }
+
+    /// <summary>
+    /// Creates an <see cref="ISKFunction"/> instance for a semantic function using the specified markdown text.
+    /// </summary>
+    /// <param name="kernel">Kernel instance</param>
+    /// <param name="text">YAML representation of the <see cref="PromptModel"/> to use to create the semantic function</param>
+    /// <param name="functionName">The function name</param>
+    /// <param name="pluginName">The optional name of the plug-in associated with this method.</param>
+    /// <param name="promptTemplateFactory">>Prompt template factory.</param>
+    /// <param name="loggerFactory"></param>
+    /// <returns>The created <see cref="ISKFunction"/>.</returns>
+    public static ISKFunction FromMarkdown(
```

Comment:
What `From` ? 

Although we have the description in the XML, it's not clear the intent of what we are getting from, by default I would assume (this, the Kernel) is what I'm getting from a Markdown, which is not the case.

This only creates an instance or does import it also to the Kernel?
If `Yes`, `ImportFunction` would be my preferred prefix.

Suggestion: Include the Function prefix (Apply for others)
```suggestion
    public static ISKFunction FunctionFromMarkdown(
    //OR
    public static ISKFunction ImportFunctionFromMarkdown(
```